File: app/src/test/java/com/neocaptainnemo/testing/service/AddressFormatterTest.java
package com.neocaptainnemo.testing.service;

import com.neocaptainnemo.testing.model.AtmNode;
import com.neocaptainnemo.testing.model.Tag;
import com.neocaptainnemo.testing.service.AddressFormatter;

import org.assertj.core.api.Assertions;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.junit.MockitoJUnitRunner;

@RunWith(MockitoJUnitRunner.class)
public class AddressFormatterTest {

    AddressFormatter formatter = new AddressFormatter();

    @Test
    public void hasCityHouseNumberStreetZip() {

        AtmNode atmNode = new AtmNode();
        Tag tag = new Tag();
        tag.setCity("City");
        tag.setHouseNumber("8");
        tag.setPostCode("12345678");
        tag.setStreet("Street");

        atmNode.setTag(tag);
        Assertions.assertThat(formatter.format(atmNode)).isEqualTo("Street 8, 12345678, City");
    }

    @Test
    public void hasCity() {

        AtmNode atmNode = new AtmNode();
        Tag tag = new Tag();
        tag.setCity("City");

        atmNode.setTag(tag);
        Assertions.assertThat(formatter.format(atmNode)).isEqualTo("City");
    }

    @Test
    public void hasCityHouseNumberStreet() {

        AtmNode atmNode = new AtmNode();
        Tag tag = new Tag();
        tag.setCity("City");
        tag.setHouseNumber("8");
        tag.setStreet("Street");

        atmNode.setTag(tag);
        Assertions.assertThat(formatter.format(atmNode)).isEqualTo("Street 8, City");
    }

    @Test
    public void hasHouseNumberStreetZip() {

        AtmNode atmNode = new AtmNode();
        Tag tag = new Tag();
        tag.setHouseNumber("8");
        tag.setPostCode("12345678");
        tag.setStreet("Street");

        atmNode.setTag(tag);
        Assertions.assertThat(formatter.format(atmNode)).isEqualTo("Street 8, 12345678");
    }

    @Test
    public void hasCityStreetZip() {

        AtmNode atmNode = new AtmNode();
        Tag tag = new Tag();
        tag.setCity("City");
        tag.setPostCode("12345678");
        tag.setStreet("Street");

        atmNode.setTag(tag);
        Assertions.assertThat(formatter.format(atmNode)).isEqualTo("Street, 12345678, City");
    }

    @Test
    public void hasCityStreet() {

        AtmNode atmNode = new AtmNode();
        Tag tag = new Tag();
        tag.setCity("City");
        tag.setStreet("Street");

        atmNode.setTag(tag);
        Assertions.assertThat(formatter.format(atmNode)).isEqualTo("Street, City");
    }


    @Test
    public void hasNothing() {

        AtmNode atmNode = new AtmNode();
        Tag tag = new Tag();

        atmNode.setTag(tag);
        Assertions.assertThat(formatter.format(atmNode)).isEmpty();
    }
}

File: app/src/test/java/com/neocaptainnemo/atms/service/AddressFormatterTest.kt
package com.neocaptainnemo.atms.service

import com.neocaptainnemo.atms.model.AtmNode
import com.neocaptainnemo.atms.model.Tag
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.junit.MockitoJUnitRunner

@RunWith(MockitoJUnitRunner::class)
class AddressFormatterTest {

    var formatter = AddressFormatter()

    @Test
    fun hasCityHouseNumberStreetZip() {

        val atmNode = AtmNode()
        val tag = Tag()
        tag.city = "City"
        tag.houseNumber = "8"
        tag.postCode = "12345678"
        tag.street = "Street"

        atmNode.tags = tag
        assertThat(formatter.format(atmNode)).isEqualTo("Street 8, 12345678, City")
    }

    @Test
    fun hasCity() {

        val atmNode = AtmNode()
        val tag = Tag()
        tag.city = "City"

        atmNode.tags = tag
        assertThat(formatter.format(atmNode)).isEqualTo("City")
    }

    @Test
    fun hasCityHouseNumberStreet() {

        val atmNode = AtmNode()
        val tag = Tag()
        tag.city = "City"
        tag.houseNumber = "8"
        tag.street = "Street"

        atmNode.tags = tag
        assertThat(formatter.format(atmNode)).isEqualTo("Street 8, City")
    }

    @Test
    fun hasHouseNumberStreetZip() {

        val atmNode = AtmNode()
        val tag = Tag()
        tag.houseNumber = "8"
        tag.postCode = "12345678"
        tag.street = "Street"

        atmNode.tags = tag
        assertThat(formatter.format(atmNode)).isEqualTo("Street 8, 12345678")
    }

    @Test
    fun hasCityStreetZip() {

        val atmNode = AtmNode()
        val tag = Tag()
        tag.city = "City"
        tag.postCode = "12345678"
        tag.street = "Street"

        atmNode.tags = tag
        assertThat(formatter.format(atmNode)).isEqualTo("Street, 12345678, City")
    }

    @Test
    fun hasCityStreet() {

        val atmNode = AtmNode()
        val tag = Tag()
        tag.city = "City"
        tag.street = "Street"

        atmNode.tags = tag
        assertThat(formatter.format(atmNode)).isEqualTo("Street, City")
    }


    @Test
    fun hasNothing() {

        val atmNode = AtmNode()
        val tag = Tag()

        atmNode.tags = tag
        assertThat(formatter.format(atmNode)).isEmpty()
    }
}
